vision/obstacle/obstacle_tracker.py

```diff
@@ -0,0 +1,95 @@
+"""
+Keeps track of detected obstacles between frames
+"""
+
+import cv2
+import numpy as np
+from vision.bounding_box import BoundingBox
+
+
+class Obstacle:
+    """
+    Properties of an obstacle in the frame.
+
+    Parameters
+    ----------
+    bbox: BoundingBox
+        bounding box of obstacle 
+    """
+    def __init__(self, bbox: BoundingBox):
+        self.bounding_box = bbox
+        self.center = np.mean(bbox)
+        self.frames_persisted = 0
+
+
+class ObstacleTracker:
+    """
+    Track obstacles between frames.
+    """
+    def __init__(self):
+        self.MVMT_TOLERANCE = 0.10 # bounding box displacement tolerance per frame
+        self.PERSISTENCE_THRESHOLD = 5 # number of frames for an obstacle to be considered consequential
+        
+        self.obstacles = np.array([]) # buffer that stores current Obstacles in view
+
+    def _isSameObstacle(self, obj1: Obstacle, obj2: Obstacle) -> bool:
+        """
+        Returns True if the position of obj1 is within self.MVMT_TOLERANCE of obj2 
+
+        Parameters
+        ----------
+        obj1: Obstacle
+            The obstacle of reference
+        obj2: Obstacle
+            The obstacle in question
+        """
+        return obj1.center >= (1-self.MVMT_TOLERANCE)*obj2.center and obj1.center <= (1+self.MVMT_TOLERANCE)*obj2.center

```

Comment:
Since the center has both an x and a y value, you need to compare x and y separately, then logical and the results from each.